### ISSUE TITLE
test explorer not loading for a huge project with 50+ packages

### ISSUE THREAD

**Issue by jgheewala** [ppparihar/GoTestExplorer#6]
Hi,

I have a huge project with 50+ packages. There are almost 500+ test cases. I installed go test explorer and waited for almost 20 mins, but nothing got loaded. I would prefer having a setting flag which indicates what package/dir you want to load into gotest explorer.

**Comment by ppparihar** [comment]
@jgheewala It looks in srs folder only If a project contains src folder at root level else it looks for all the folder/directory. The extension has tested with 300+ tests successfully. If possible can you share your git repo? else please share your directory structure?
I will love to work on it. Feel free to raise queries/suggestions.

**Comment by ppparihar** [comment]
@jgheewala are you still facing the same issue?

**Comment by jgheewala** [comment]
@ppparihar yes I am still facing the issue. Unfortunately i cannot share private repo access from the company. So i think I know what the issue was. 

Temporary Fix for below:

- I had 80+ different golang projects including Shopify/saram & golang/x/net in my GOPATH/src/ folder.. 
- I removed third-party projects which got included as part of "go get" cmd execution.
- now i can see my project test cases

Enhancement request:

Provide a config for user to setup a list of package paths and display test cases only from this list. If no elements then preserve the default behavior.

**Comment by gfrancqu** [comment]
Hi, 

I'm facing the same issue, the fact is that my GOPATH is located inside my repo,
the testExplorer actually [only skip the .vscode, .git and vendor ](https://github.com/ppparihar/GoTestExplorer/blob/4cbbac32d13397e0edbe21a269a0ff4be8517e2d/src/config.ts#L4)

maybe it could access and use the workspace settings `files.exclude` ?
